pkg/image/fetcher.go

```diff
@@ -123,6 +123,37 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 	return f.fetchDaemonImage(name)
 }
 
+func (f *Fetcher) CheckReadAccessValidator(repo string, options FetchOptions) bool {
+	if !options.Daemon {
+		return f.checkRemoteReadAccess(repo)
+	}
+	if _, err := f.fetchDaemonImage(repo); err != nil {
+		// Image doesn't exist in the daemon
+		// 	Pull Never: should failed
+		// 	Pull Always or Pull If Not Present: need to check the registry
+		if options.PullPolicy == PullNever {
+			return false
+		}
+		return f.checkRemoteReadAccess(repo)
+	}
+	// no-op
+	return true
+}
+
+func (f *Fetcher) checkRemoteReadAccess(repo string) bool {
+	img, err := remote.NewImage(repo, f.keychain)
+	if err != nil {
+		return false
```

Comment:
Don't we potentially hide an error here?

Reply:
Maybe, I just added a debug message into the console